examples/30_extended/fetch_runtimes_tutorial.py

```diff
@@ -0,0 +1,281 @@
+# *******************************************************
+# We'll run a Random Forest model and obtain an OpenML run object. We can
+# see the evaluations recorded per fold for the dataset and the information
+# available for this run.
+clf = RandomForestClassifier(n_estimators=10)
+
+run1 = openml.runs.run_model_on_task(
+    model=clf, task=task, upload_flow=False, avoid_duplicate_runs=False,
+)
+measures = run1.fold_evaluations
+
+# The timing and performance metrics available are
+for key in measures.keys():
+    print(key)
+
+# The performance metric is recorded under `predictive_accuracy` per fold
+# and can be retrieved as:
+for repeat, val1 in measures["predictive_accuracy"].items():
+    for fold, val2 in val1.items():
+        print("Repeat #{}-Fold #{}: {:.4f}".format(repeat, fold, val2))
+    print()
+
+################################################################################
+# The remaining entries recorded in `measures` are the runtime records
+# related as:
+# usercpu_time_millis = usercpu_time_millis_training + usercpu_time_millis_testing
+# wall_clock_time_millis = wall_clock_time_millis_training + wall_clock_time_millis_testing
+#
+# The timing measures recorded as `*_millis_training` contain the per
+# repeat-fold timing incurred for the executing of the `.fit()` procedure
+# of the model. For `usercpu_time_*` the time recorded using `time.process_time()`
+# is converted to milliseconds and stored. Similarly, `time.time()` is used
+# to record the time entry for `wall_clock_time_*`. The `*_millis_testing` entry
+# follows the same procedure but for time taken for the `.predict()` procedure.
+
+# Comparing the CPU and wall-clock training times of the Random Forest model
+print_compare_runtimes(measures)
+
+################################################################################
+# Case 2: Running a Random Forest model on an OpenML task in parallel (2 cores)
+# *****************************************************************************
+# Redefining the model to allow parallelism with `n_jobs=2`
+clf = RandomForestClassifier(n_estimators=10, n_jobs=2)
+
+run2 = openml.runs.run_model_on_task(
+    model=clf, task=task, n_jobs=4, upload_flow=False, avoid_duplicate_runs=False,
+)
+measures = run2.fold_evaluations
+# The wall-clock time recorded per fold should be lesser than Case 1 above
+print_compare_runtimes(measures)
+
+####################################################################################
+# Running a Random Forest model on an OpenML task in parallel (all cores available):
+
+# Redefining the model to use all available cores with `n_jobs=-1`
+clf = RandomForestClassifier(n_estimators=10, n_jobs=-1)
+
+run3 = openml.runs.run_model_on_task(model=clf, task=task, dataset_format="array")
+measures = run3.fold_evaluations
+# The wall-clock time recorded per fold should be lesser than the case above,
+# if more than 2 CPU cores are available
+print_compare_runtimes(measures)
```

Comment:
The results on my local machine are really interesting and I assume similar things will happen on CI, too:
```
Repeat #0-Fold #0: CPU-205.384 vs Wall-2071.654
Repeat #0-Fold #1: CPU-78.619 vs Wall-661.656
Repeat #0-Fold #2: CPU-63.022 vs Wall-520.443
Repeat #0-Fold #3: CPU-66.916 vs Wall-543.427
Repeat #0-Fold #4: CPU-66.177 vs Wall-488.355
Repeat #0-Fold #5: CPU-67.053 vs Wall-495.918
Repeat #0-Fold #6: CPU-65.344 vs Wall-474.387
Repeat #0-Fold #7: CPU-68.329 vs Wall-689.225
Repeat #0-Fold #8: CPU-80.341 vs Wall-637.022
Repeat #0-Fold #9: CPU-83.629 vs Wall-649.633
```
Could you please add a comment on the fact that the CPU time no longer contains the actual computation and can therefore not be used in a meaningful way?

Reply:
Done.

Reply:
Added a bit of snippet.

Reply:
But we do know the number of cores here, don't we? We have the case that subprocesses are started and they are not recorded in the CPU time, right?